exonum/src/runtime/rust/call_context.rs

```diff
@@ -0,0 +1,110 @@
+// Copyright 2019 The Exonum Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use exonum_merkledb::{BinaryValue, Fork};
+
+use crate::runtime::{
+    BlockchainMailbox, CallInfo, Caller, DispatcherRef, ExecutionContext, ExecutionError,
+    InstanceId, MethodId,
+};
+
+// TODO Write a full documentation when the interservice communications are fully implemented. [ECR-3493]
+/// Provide a low level context for the call of methods of a different service instance.
+#[derive(Debug)]
+pub struct CallContext<'a> {
```

Comment:
Eventually `CallContext` was in the `mod.rs` despite the fact that it's specific to the Rust runtime.